src/utils/error.js

```diff
@@ -0,0 +1,75 @@
+const {
+  Prisma: { PrismaClientKnownRequestError },
+} = require("@prisma/client");
+
+const Types = require("../utils/types.d.js");
+
+const errorCodes = {
+  P2002: uniqueConstraintFailed,
+  P2003: foreignKeyConstraintFailed,
+  P2025: recordNotFound,
+};
+
+/**
+ * @param {Error} error
+ * @param {Types.Response} res
+ * @returns {void}
+ */
+function handleError(error, res) {
+  if (error instanceof PrismaClientKnownRequestError) {
+    const errorHandler = errorCodes[error.code];
+    if (errorHandler) {
+      errorHandler(error, res);
+      return;
+    }
+  }
+
+  res.status(500).json({ error: error.message });
+}
+
+/**
+ * @param {import("@prisma/client").Prisma.PrismaClientKnownRequestError} error
+ * @param {Types.Response} res
+ * @returns {void}
+ */
+function uniqueConstraintFailed(error, res) {
+  let errorMessage = error.message;
+  if (error.meta && typeof error.meta.modelName === "string") {
+    const model = error.meta.modelName.toLowerCase();
+    const target = error.meta.target;
```

Comment:
Presumably I can't mix this with the `toLowerCase()` method if I destructure here?
I guess it would make more sense to do the following then?
```js
const { modelName, target } = error.meta
errorMessage = `A ${modelName.toLowerCase()} with the provided ${target} already exists`
```
Or is good english error message optional is it for developers?